src/main/java/com/gmail/goosius/siegewar/SiegeWar.java

```diff
@@ -55,15 +59,18 @@ public File getSiegeWarJarFile() {
 	public static SiegeHUDManager getSiegeHUDManager() {
 		return siegeHUDManager;
 	}
+
+	public SiegeWar() {
+		plugin = this;
+		this.scheduler = townyVersionCheck() ? isFoliaClassPresent() ? new FoliaTaskScheduler(this) : new BukkitTaskScheduler(this) : null;
+	}

```

Comment:
❔ 
- Hi, could this above logic be change to something a bit easier to follow please?

Reply:
Resolved by extracting the method and dropping the double ternary.